package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -1194,7 +1194,7 @@ def save_table(self, filename="hbond_table.pickle"):
         """
         if self.table is None:
             self.generate_table()
-        with open(filename, 'w') as f:
+        with open(filename, 'wb') as f:
             cPickle.dump(self.table, f, protocol=cPickle.HIGHEST_PROTOCOL)
```

Comment:
This is actually not good for python 2/3 compatibility. It means that a pickle written with python 3 won't be readable with python 2. What is the type of table? Is it a numpy array? Then we have more portable options.

Reply:
@kain88-de Thank you for the comment. I have tested reading and writing in py3 but only tested writing in py2.
This is a numpy recarray. I guess np.save() might be a better option?

Reply:
I would use `np.savetxt` then a user can consume the table however he likes. But this is a deprecation. So if this is read from somewhere we should allow reading a pickle for some time. Maybe even writing a pickle is OK until a version 1.0 (If the user really wants to).